### DISCUSSION TITLE
what am i doing wrong or is maxima really that worthless or could it be a bug

### DISCUSSION THREAD

**Discussion by zmth** [wxMaxima-developers/wxmaxima#1593] [Q&A]
build_info(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")build_info(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")(version="branch_5_44_base_231_g5c411f69f",timestamp="2021-01-12 23:51:42",host="x86_64-w64-mingw32",lisp_name="SBCL",lisp_version="2.0.0",maxima_userdir="C:/Users/zmth1/maxima",maxima_tempdir="C:/Users/zmth1/AppData/Local/Temp",maxima_objdir="C:/Users/zmth1/maxima/binary/branch_5_44_base_231_g5c411f69f/sbcl/2_0_0",maxima_frontend="wxMaxima",maxima_frontend_version="20.12.2-DevelopmentSnapshot_MSW_OpenMP201511+Locks")

((declare([n,k],integer),assume(n>1,n>k,k>0), (n+1)!^2/k!^2/(n+1-k)!^2- n!^2/k!^2/(n-k)!^2=((n+1)^2-(n+1-k)^2)*n!^2/((n+1-k)!^2*k!^2))``
ratsimp((n+1)!^2/k!^2/(n+1-k)!^2- n!^2/k!^2/(n-k)!^2)

hope can understand because something is really wrong with this interface as it will NOT insert code! and for some reason put a " after the first expression when it should not have.
anyway the rhs of the first expression which is over a common denominator is what I want and certainly what should
have been the same with ratsimp but it is not - not reduced to the proper form which i had to do manually as rhs of 1st
expression. I even tried factor and that did not work either. Certainly one would not have to go to this disordinate ridiculous trouble for something maxima should do itself. Prior maxima did work and do it properly though i did not have factorial expressions which certainly should make no difference. What commands should i be using ? and don't want to have to go to gamma fct esp since all is in integers. So what does one have to do to get maxima to do what it should be doing. Eg i also even tried facsum, factorfacsum still same unreduced proper output as i had to do manually.
which is

**Comment by zmth** [comment]
well what do you know ! Wonders never cease ! I will have to take back a lot of what i said anyway it may help others with the same type of issue so I won't delete it. The key seems to be : to set he following true 
factorial_expand:true  though i would have thought with that type of preferred behavior and saving it would have been default true. But anyway it came back better than my first work as i forgot to combine trivial terms
`(declare([n,k],integer),assume(n>1,n>k,k>0),factorial_expand:true, factor((n+1)!^2/k!^2/(n+1-k)!^2- n!^2/k!^2/(n-k)!^2));
**and did not have to convert anything to gamma`** another issue here nothing to do with this issue is it seems the code and bold B don't do anything or certainly not as they do in the maxima bugs forums. Also why does it say just below 'Styling with Markdown is supported' and what does that mean. If anything it seems that it isn't.

**Comment by zmth** [comment]
NOw something really unexplainable though cannot necessarily say it is a bug either so someone please explain this non-sensical error message in red. Assume factorial_expand:falseand do:
`(fp:binomial(n+1,k)^2/ binomial(2*n+2,n+1),f: binomial(n,k)^2/binomial(2*n,n), dg:(-3*n+2*k-1)*n!^2/(2*n+1)/k!^2/(n-k)!^2/binomial(2*n,n)/2-(-3*n+2*k-3)*n!^2/(2*n+1)/(k-1)!^2/(n-k+1)!^2/binomial(2*n,n)/2,for n_:2 thru 4 do for k_ thru n_-1 do(t:lratsubst([n=n_,k=k_],fp-f-dg) ,if t#0 then(fl:false,disp(["n",n_,"k",k_,t]))),fl);`

**And it comes back with no error as expected and true** but now if i preface the above with 
factorial_expand:true
it comes back with a red error "factorial: factorial of negative integer -1 not defined....." which certainly seems like a bogus error. If not please someone explain this irrational seemingly out of nowhere for NO reason one gets the red error !

**Reply by zmth** [reply]
please change the true and false cases I said it just backwards . Too bad there is no way to edit it. REally want to make that feature request ! or do i have to go to github ? wish could get out of these 2 different whatever and keep it all maxima with NOTHIng to do with GIT hub or whatever. so so confusing !! like for example what in the hell is Markdown for one. Does anyone have any idea what Markdown means ?

**Reply by tigran123** [reply]
Markdown is a language that you use to communicate with humans using github. Each interface on this planet (living or otherwise) has its own language, e.g. to plot functions you have to learn gnuplot language, to write really fast programs you have to learn C language, to write really useful programs you have to learn python language, to write programs that nobody (including yourself) will understand a few months after they are written you have to learn perl language and so on :)

**Reply by zmth** [reply]
except i would change that to ....to write really fast programs you have to use fortran 2008 which is a hell of lot better than that damn difficult C. And fortan is much faster than python and why on earth this perl crap ? Anyway for more confusion where does regular expression search which is pretty poor to me and seems one could really do all with Fortran 2008 but getting off subject and personal preference. So anyway is this setup and meanings in this ...discussion come from all maxima's choice of standards and definition or is it github or part both. I really don't see the meaning of all of this and WHY is there NO really needed 'edit' function and whose fault is it that such does NOT exist ? thanks . there still seems to be quite a bit else including all those H B slanted I..... about 13 different ones in all and why don't they do the same as they do in bugs. Then again at the very top of it all is < > code again along with issues , Pull requests , Discussions, Actions , Projects, Wiki , Security , and god knows Insights . I just sent github a rather complaining email about all this non exact questionable meanings of this and that and how all is supposed to fit together. So did i do the right thing or is it maximas issues and definitions ? Still the most important and needed issue is an Edit function such as in math stack xchg and all which seems much better organized than all this and that jazz here and in bugs another discussion forum which does not seem to do anything other than to say no forums exist, and then there is wxmaxima etc. etc. and still how to make any use of Styling with Markdown is supported. yea ok so what is one supposed to make any use of that even if it is. and then there is write,mark as answer , suggest as answer, write a reply and then labels,create issue from discussion, another up arrow a small vertical bar etc. So how is one to know the proper procedure to use from all this hodge podge of... whatever ?

**Reply by tigran123** [reply]
The answer to any question like "how is one to know ..." is ever the same: "knowledge comes only by learning, it is protected by sharing, made real and useful by wisdom and socialised by love". Seriously, there is a button labelled `M Styling with Markdown is supported` below this edit window -- try clicking on it and read what ensues.

**Reply by zmth** [reply]
yea i hear u. Wonder why it is that everyone else seems to know exactly what to do and no question. Anyway i understand it would be way too time consuming to try and explain all these questions at once and all this information is NOT in the maxima help file?. But anyway for right now bottom line is since there is no edit one has to erase all and just start all again i suppose ? thanks

**Reply by zmth** [reply]
yea clicking on that styling...markdown .... is quite a lot of information which seems to tell me it is a github issue rather than maxima

**Reply by tigran123** [reply]
No, that is not the bottom line at all. That is just an assumption, based on the premise "I have learnt all there is to learn about this particular interface" and both happen to be false. There certainly IS a way to edit the message. Just click on the `Edit` link in the menu.

**Reply by zmth** [reply]
yea i just found it just before reading your last comment so that IS quite a relief ! enough so now can get back to waiting for the real issue at hand which is the strictly maxima error message issue though i seem to have answered the main deal myself being the 
factorial_expand:true  or false switch. if/when have time by and by will try to learn about all these other 'do dads' without complaints and 'jumping the gun' which admitedly could be irritating to others.
